src/OSInfo.cpp

```diff
@@ -22,6 +22,86 @@
 #include "OSInfo.h"
 #include "Version.h"
 
+#if defined(Q_OS_WIN)
+/// Query for a Windows 10-style displayable version.
+///
+/// This returns a string of the kind:
+///
+///    Windows 10 Pro x86 1607 14390.0
+///
+/// which is:
+///
+///    $ProductName $Arch $Version $Build.$Ubr
+///
+/// Of note, $Version is formatted as YYMM.
+/// So, 1607 is a Windows 10 update released in July 2016.
+///
+/// This function can be called on non-Windows 10 OSes.
+/// On those, this functions fails to query some of the
+/// registry keys used for building the version string
+/// to be returned. Because of that, it is safe to call
+/// this function, and if it returns an empty/null string,
+/// a legacy version string can be displayed.
+static QString win10DisplayableVersion() {
```

Comment:
In any case, it's just a style thing, I suppose.

I like headers to int expose the public API. I made it a static function to avoid polluting the header.

I'm open to suggestions why to prefer one over the other.